src/views/SystemProfile/components/FundingAndBudget/index.test.tsx

```diff
@@ -1,24 +1,88 @@
 import React from 'react';
 import { MemoryRouter, Route } from 'react-router-dom';
-import { render } from '@testing-library/react';
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import { cloneDeep } from 'lodash';
 
-import { getMockSystemProfileData } from 'data/mock/systemProfile';
+import { BUDGET_ITEMS_COUNT_CAP } from 'constants/systemProfile';
+import { getMockSystemProfileData, result } from 'data/mock/systemProfile';
+// eslint-disable-next-line camelcase
+import { GetSystemProfile_cedarBudget } from 'queries/types/GetSystemProfile';
 
 import FundingAndBudget from './index';
 
 const systemProfileData = getMockSystemProfileData();
 
-describe('Funding and Budget subpage for System Profile', () => {
-  it('matches snapshot', async () => {
-    const { asFragment, getByText } = render(
+describe(`System Profile Funding and Budget section collapse/expand toggle at ${BUDGET_ITEMS_COUNT_CAP}`, () => {
+  const buttonExpandToggleMatchOpt = {
+    name: /Show more budget projects/i
+  };
+
+  function fillBudgetItems(
+    count: number
+    // eslint-disable-next-line camelcase
+  ): GetSystemProfile_cedarBudget[] {
+    return Array(count)
+      .fill(0)
+      .map((_, index) => {
+        return {
+          fiscalYear: '',
+          funding: '',
+          fundingId: '',
+          fundingSource: '',
+          id: '',
+          name: '',
+          projectId: '12345',
+          projectTitle: '',
+          systemId: '',
+          __typename: 'CedarBudget'
+        };
+      });
+  }
+
+  it(`doesn't show toggles for all sections`, () => {
+    render(
       <MemoryRouter initialEntries={['/systems/000-100-0/funding-and-budget']}>
         <Route path="/systems/:systemId/:subinfo">
           <FundingAndBudget system={systemProfileData} />
         </Route>
       </MemoryRouter>
     );
-    expect(asFragment()).toMatchSnapshot();
-    expect(getByText('HAM Project funding')).toBeInTheDocument();
-    expect(getByText('Fed Admin')).toBeInTheDocument();
+
+    // Doesn't show the button expand toggle at cap
+    const data = getMockSystemProfileData(result.data);
+    const { queryAllByRole } = render(<FundingAndBudget system={data} />);
```

Comment:
Post about using `screen` methods
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen

Reply:
thank you for linking that article - seems super helpful, excited to read through it.